README.md

```diff
@@ -11,54 +11,437 @@
-Wallet, which has been completely rewritten for the
-[Shelley](https://cardanoroadmap.com/) phase.
+### Docker
 
-The Byron version of Cardano Wallet is in the
-[cardano-sl](https://github.com/input-output-hk/cardano-sl)
-repository.
+Use the following commands to run the cardano-wallet and cardano-node at the same time using docker:
 
-## How to build
+### Testnet
+To start the docker containers for cardano-wallet and cardano-node:
+```console
+$ NETWORK=testnet TESTNETFILE=/app/config/testnet-genesis.yaml docker-compose up
+```
+
+Note that `TESTNETFILE` is required when running the above command, this is only for the `testnet` environment.
 
-Use [Haskell Stack](https://haskellstack.org/) to build this project:
+You can verify that it is running:
 
+```console
+$ curl http://localhost:8090/v2/network/information
+{
+  "network_tip": {
+    "epoch_number": 57,
+    "slot_number": 1929
+  },
+  "node_tip": {
+    "height": {
+      "quantity": 50829,
+      "unit": "block"
+    },
+    "epoch_number": 2,
+    "slot_number": 8659
+  },
+  "sync_progress": {
+    "status": "syncing",
+    "progress": {
+      "quantity": 4.13,
+      "unit": "percent"
+    }
+  },
+  "next_epoch": {
+    "epoch_start_time": "2020-05-09T20:20:16Z",
+    "epoch_number": 58
+  }
+}
 ```
-$ stack build --test --no-run-tests
+To stop the docker containers:
+```console
+$ docker-compose down
 ```
 
-## How to test
+### Mainnet
+To start the docker containers for cardano-wallet and cardano-node:
+```console
+$ NETWORK=mainnet docker-compose up
+```
+Check it is running by making a request.
+```console
+$ curl http://localhost:8090/v2/network/parameters/latest
+{
+  "slot_length": {
+    "quantity": 20,
+    "unit": "second"
+  },
+  "epoch_stability": {
+    "quantity": 2160,
+    "unit": "block"
+  },
+  "genesis_block_hash": "5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb",
+  "blockchain_start_time": "2017-09-23T21:44:51Z",
+  "epoch_length": {
+    "quantity": 21600,
+    "unit": "slot"
+  },
+  "active_slot_coefficient": {
+    "quantity": 100,
+    "unit": "percent"
+  }
+}
+
+```
+To stop the docker containers:
+```console
+$ docker-compose down
+```
+
+### Switching from one environment to another
+When switching between running testnet on docker to mainnet on docker, you should clear the volumes using the commands below or alter the docker-compose file to have persistent volumes mapped to your drive outside of the docker container.  
+NB! It is imperative that you keep wallet recovery information basked-up outside of the docker volumes as you may choose to remove the volumes at some stage. Wallets should be restored via the API as needed. You will receive an ID back when you restore a wallet. You can also list the active wallets managed by cardano-wallet.
+
+Use with caution!
+```console
+$ docker-compose down
+$ docker volume rm cardano-wallet_node-ipc
+$ docker volume rm cardano-wallet_node-db
+```
```

Comment:
This and the 3 sections above would better be in the Wiki. README is best kept short, pointing to other points of the documentation.